pkg/controller/configmap/config.go

```diff
@@ -571,6 +578,29 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 	return needChange, nil
 }
 
+func immutableFieldChanged(cur, prev *ini.File) bool {
+	immutableFields := []struct {
+		Section string
+		Key     string
+	}{
```

Comment:
Do we need a const dict to save the options?

Reply:
Golang does not support constant arrays, maps or slices, and I think we can use global variable instead.